lib/rules/selector-not-notation/__tests__/index.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const { messages, ruleName } = require('..');
+
+testRule({
+	ruleName,
+	config: ['simple'],
+	fix: false,
+
+	accept: [
+		{
+			code: ':not() {}',
+		},
+		{
+			code: ':not( div ) {}',
+		},
+		{
+			code: ':nOt(div) {}',
+		},
+		{
+			code: ':not(*) {}',
+		},
+		{
+			code: ':not(:link) {}',
+		},
+		{
+			code: ':not(.foo) {}',
+		},
+		{
+			code: ':not([title]) {}',
+		},
+	],
+
+	reject: [
+		{
+			code: ':not(:not()) {}',
+			fixed: ':not()',
+			message: messages.expected('simple'),
+			line: 1,
+			column: 1,
+		},
+		{
+			code: ':not(:before) {}',
+			fixed: ':not()',
+			message: messages.expected('simple'),
+			line: 1,
+			column: 1,
+		},
+		{
+			code: ':not(input, select) {}',
+			fixed: ':not(input):not(select) {}',
+			message: messages.expected('simple'),
+			line: 1,
+			column: 1,
+		},
+		{
+			code: ':not(p.foo) {}',
+			fixed: '' /* how to fix this one? */,
```

Comment:
e.g.
`:not(p)` discards _all_ p tags hence `:not(p):not(.foo)` would not match any p
`:not(p.foo)` may (if there are p left without a foo class)